gpytorch/lazy/lazy_evaluated_kernel_tensor.py

```diff
@@ -251,3 +255,16 @@ def representation_tree(self):
         # representation
         else:
             return self.evaluate_kernel().representation_tree()
+
+    def __getitem__(self, index):
+        """
+        Supports subindexing of the matrix this LazyTensor represents. This may return either another
+        :obj:`gpytorch.lazy.LazyTensor` or a :obj:`torch.tensor` depending on the exact implementation.
+        """
+        # Process the index
+        index = index if isinstance(index, tuple) else (index,)
+        if index[0] is Ellipsis and all([isinstance(idx, slice) for idx in index[1:]]):
+            *batch_indices, row_index, col_index = index
+            return self._getitem(row_index, col_index, *batch_indices)
```

Comment:
This could have weird issues if `len(index) != self.dim()`.

Reply:
Made it more explicitly a special case for indexing of the form `mat[..., slice, slice]`.

Reply:
This could also still fail if the index is `mat[..., 0:3]`, for example.